src/middlewared/middlewared/plugins/container_runtime_interface/images.py

```diff
@@ -117,66 +115,34 @@ async def pull(self, job, data):
 
         `docker_authentication` should be specified if image to be retrieved is under a private repository.
         """
-        await self.docker_checks()
-        # TODO: Have job progress report downloading progress
-        async with aiodocker.Docker() as docker:
+        self.middleware.call_sync('kubernetes.validate_k8s_setup')
+        with ContainerdClient('image') as client:
             try:
-                response = await docker.images.pull(
-                    from_image=data['from_image'], tag=data['tag'], auth=data['docker_authentication']
+                client.pull_image(
+                    f'{data["from_image"]}:{data["tag"]}', auth_config=data['docker_authentication'],
```

Comment:
Docker is basically "dying", it might be best to start using non-branded terminology for things like these internally at least.